lib/module.php

```diff
@@ -629,3 +635,90 @@ function solusiovps_restart(array $params)
         return $e->getMessage();
     }
 }
+
+function solusiovps_ListAccounts(array $params)
+{
+    try {
+        $statusMap = [
+            "0" => WHMCS\Service\Status::ACTIVE,
+            "1" => WHMCS\Service\Status::SUSPENDED,
+        ];
+        $accounts = [];
+
+        $serverId = (int)$params['serverid'];
+        $serverParams = Server::getParams($serverId);
+        $serverResource = new ServerResource(Connector::create($serverParams));
+
+        $current_page = 1;
+        do {
+            $serversOnPage = $serverResource->listServerPage($current_page);
+            $lastPage = Arr::get($serversOnPage, 'meta', [])['last_page'];
+
+            foreach (Arr::get($serversOnPage, 'data', []) as $server) {
+                $accounts[] = [
+                    'email' => $server['user']['email'],
+                    'username' => $server['user']['email'],
+                    'domain' => $server['name'],
+                    'uniqueIdentifier' => $server['name'],
+                    'product' => $server['plan']['name'],
+                    'primaryip' => $server['ip_addresses']['ipv4'][0]['ip'],
+                    'created' => Carbon::parse($server['created_at'])->format('Y-m-d H:i:s'),
+                    'status' => $statusMap[$server['is_suspended']],
+                ];
+            }
+            $current_page++;
+        } while ($current_page <= $lastPage);
+
+        return [
+            'success' => true,
+            'accounts' => $accounts,
+        ];
+    } catch (Exception $e) {
+        return [
+            'success'  => false,
+            'error' => $e->getMessage(),
+        ];
+    }
+}
+
+function solusiovps_syncAccount(array $params)
+{
+    try {
+        if (!empty(SolusServer::getByServiceId($params['serviceid']))) {
+            return [
+                'success' => "Account is already synced",
+            ];
+        }
+
+        $serverId = (int)$params['serverid'];
+        $serverParams = Server::getParams($serverId);
+
+        $userResource = new UserResource(Connector::create($serverParams));
+        $solusUser = $userResource->getUserByEmail($params['clientsdetails']['email']);
+
+        $serverResource = new ServerResource(Connector::create($serverParams));
+        $allServersOfUser = $serverResource->getAllByUser($solusUser['id']);
+
+        foreach ($allServersOfUser as $server) {
+            if ($server['name'] == $params['domain']) {
+                $serverPayload = $serverResource->get($server['id']);
```

Comment:
Can't we take serverPayload directly from the $server, if I correctly understand they have the same contract.

Reply:
Thanks a lot for the input! I have adjusted the code according to your comments.